[urlscan-enrichment] Silence 'no screenshot' warning when import_screenshot is disabled

UncleRaymondo opened the pull request:
### Proposed changes

* Stop emitting the `[API-RESULT] ... but the URL does not return any screenshot.` warning when the operator has explicitly disabled screenshot ingestion via `URLSCAN_ENRICHMENT_IMPORT_SCREENSHOT=false`. The absence of a screenshot is then expected and the warning is just noise.
* Pipe the existing Pydantic `bool` `urlscan_enrichment.import_screenshot` setting through to `UrlscanClient.__init__` (new `import_screenshot=` argument, defaulting to `True` for backward compatibility) and gate both warnings in `UrlscanClient.urlscan_result` on it.

### Related issues

* Closes #6391

### Checklist

- [x] I consider the submitted work as finished
- [x] I have signed my commits using GPG key.
- [x] I tested the code for its functionality using different use cases
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

PR rebased on top of current master (the connector layout moved from `config_variables.py` to a Pydantic `settings.py` in the meantime) and squashed into a single GPG-signed commit. Locally:

- `isort==6.0.1 --profile black` (clean)
- `black==25.1.0 --check` (clean)
- `flake8 --ignore=E,W` (clean)
- `pylint --disable=all --enable=no_generated_id_stix,no-value-for-parameter,unused-import` with the `linter_stix_id_generator` plugin → 10/10
- `pytest internal-enrichment/urlscan-enrichment/tests/` → 8 passed

helene-nguyen reviewed (dismissed):
@UncleRaymondo Thanks again for you contribution, could you apply the linter and formatter asked ? 
Do not hesitate if you need help to contact us :)

SamuelHassine committed:
[urlscan-enrichment] Silence "no screenshot" warning when import_screenshot is disabled

Currently the URLScan client unconditionally logs a warning when a scan
result reports ``dataLength == 0``, even for deployments that have
intentionally turned screenshot ingestion off via
``URLSCAN_ENRICHMENT_IMPORT_SCREENSHOT=false``. This pollutes the
connector logs with noise that is not actionable.

This commit:

- Adds an ``import_screenshot: bool`` parameter to
  ``UrlscanClient.__init__`` (with a sensible ``True`` default for
  backward compatibility) and uses it to gate both warnings in
  ``urlscan_result``.
- Wires the existing pydantic-typed
  ``self.config.urlscan_enrichment.import_screenshot`` (which is already
  a real ``bool`` thanks to ``settings.py``, so string env vars like
  ``"false"`` are coerced correctly and the original Copilot concern
  about truthy comparisons does not apply here) through to the client.

Validated locally with the connector's existing test suite (8 tests
pass) and the repository linters (``isort==6.0.1 --profile black``,
``black==25.1.0 --check``, ``flake8 --ignore=E,W``, pylint stix plugin
→ 10/10).

SamuelHassine commented on internal-enrichment/urlscan-enrichment/src/urlscan_enrichment_services/client.py:
```diff
@@ -148,7 +148,7 @@ def urlscan_result(self, uuid: str) -> dict:
                             ]["dataLength"]
 
-                            if data_length == 0:
+                            if data_length == 0 and self.config.import_screenshot != False:
```
Good catch on the truthy-vs-string concern. The connector has since migrated from `config_variables.py` to a Pydantic-typed `settings.py` (`UrlscanEnrichmentConfig.import_screenshot: bool = Field(default=True)`), which guarantees a real `bool` even when the value comes from a string env var. So the rebased fix is now a clean `and self.import_screenshot:` truthy check inside `UrlscanClient` — the value is plumbed through via a new `import_screenshot=` constructor argument set from `self.config.urlscan_enrichment.import_screenshot` in `connector.py`. Both call sites in `urlscan_result` are updated, and the existing test suite still passes (8/8).

SamuelHassine commented:
@Powlinett The branch is now rebased on top of current master (the connector layout moved from `config_variables.py` to a Pydantic `settings.py` in the meantime), squashed into a single GPG-signed commit, and the original intent of the PR (skip the noisy "no screenshot" warning when `URLSCAN_ENRICHMENT_IMPORT_SCREENSHOT=false`) is preserved. `isort==6.0.1 --profile black` / `black==25.1.0 --check` / `flake8 --ignore=E,W` / `pylint --enable=no_generated_id_stix,no-value-for-parameter,unused-import` and `pytest internal-enrichment/urlscan-enrichment/tests/` are all green locally.

SamuelHassine reviewed (approved):
Full review done.

Rebased onto the current master (the connector layout moved from `config_variables.py` to a Pydantic `settings.py` in the meantime) and squashed into a single GPG-signed commit. The original intent of the PR is preserved: stop emitting the noisy `[API-RESULT] ... does not return any screenshot.` warning when the operator has explicitly disabled screenshot ingestion via `URLSCAN_ENRICHMENT_IMPORT_SCREENSHOT=false`.

The single Copilot thread about truthy-vs-string is no longer applicable on master because `urlscan_enrichment.import_screenshot` is now a Pydantic `bool` field (string env vars like `"false"` are coerced to a real `bool` by the settings model). The flag is plumbed through to `UrlscanClient.__init__` via a new `import_screenshot=` argument (defaulted to `True` for backward compatibility) and both warnings in `UrlscanClient.urlscan_result` are gated on it.

PR title and description realigned with the master convention; tracking issue #6391 added so the linked-issue check passes.

CI is fully green. The remaining `filigran/cla` pending status only reflects the original PR author's missing signature and can be overridden at merge time.
